crates/ark/src/reticulate.rs

```diff
@@ -0,0 +1,145 @@
+use std::sync::Mutex;
+
+use amalthea::comm::comm_channel::CommMsg;
+use amalthea::comm::event::CommManagerEvent;
+use amalthea::socket::comm::CommInitiator;
+use amalthea::socket::comm::CommSocket;
+use anyhow::anyhow;
+use crossbeam::channel::Sender;
+use harp::RObject;
+use lazy_static::lazy_static;
+use libr::R_NilValue;
+use libr::SEXP;
+use serde_json::json;
+use stdext::spawn;
+use stdext::unwrap;
+use uuid::Uuid;
+
+use crate::interface::RMain;
+
+lazy_static! {
+    static ref RETICULATE_COMM_ID: Mutex<Option<String>> = Mutex::new(None);
+}
+
+pub struct ReticulateService {
+    comm: CommSocket,
+    comm_manager_tx: Sender<CommManagerEvent>,
+}
+
+impl ReticulateService {
+    fn start(comm_id: String, comm_manager_tx: Sender<CommManagerEvent>) -> anyhow::Result<String> {
+        let comm = CommSocket::new(
+            CommInitiator::BackEnd,
+            comm_id.clone(),
+            String::from("positron.reticulate"),
+        );
+
+        let service = Self {
+            comm,
+            comm_manager_tx,
+        };
+
+        let event = CommManagerEvent::Opened(service.comm.clone(), serde_json::Value::Null);
+        unwrap!(service.comm_manager_tx.send(event), Err(e) => {
```

Comment:
If return value is not used, better use the standard `if let Err(err)`

Reply:
Used `.or_log_error()` instead